docs/source/conf.py

```diff
@@ -385,6 +385,12 @@ def generate_weights_table(module, table_name, metrics, include_pattern=None, ex
 generate_weights_table(
     module=M.detection, table_name="detection", metrics=[("box_map", "Box MAP")], exclude_pattern="Keypoint"
```

Comment:
I now need to ignore 2 exclude_pattern "Mask" and "Keypoint". This will need additional refactor

Reply:
yeah, l guess we should change `accept_pattern` and `exclude_pattern` to take lists instead:

```py
    if include_patterns is not None:
        weights = [w for w in weights if any(p in str(w) for p in include_patterns)]
    if exclude_patterns is not None:
        weights = [w for w in weights if all(p not in str(w) for p in exclude_patterns)]
```

(and add an `s` to their name)

Reply:
I wish I know these python tricks. I wrote loops and logic to do same. (too much java spoils python skills)
Anyways if some day we revisit 
```
if exclude_pattern is not None:
        weights_l = []
        for w in weights:
            fg = 0
            for e in exclude_pattern:
                if e in str(w):
                    fg = 1
                    break
            if(fg == 0):
                weights_l.append(w)
        weights = weights_l
```

i will refactor as your comment :)

Reply:
> too much java spoils python skills

😅

Years ago, this thread was quite useful to me to up my Python game: https://stackoverflow.com/questions/2573135/python-progression-path-from-apprentice-to-guru

I think it's still relevant today (some of them are not, typically `map`, `reduce`, `filter` are mostly replaced by list comprehension or other utilities like `sum` or `any`, but they're staples of functional programming that are worth knowing. Same with `xrange` which is irrelevant in Python3).

Reply:
Thank you for the link :) Will read and learn more :rocket: